src/js/game/GameController.js

```diff
@@ -1341,13 +1341,14 @@ class GameController {
     if (this.levelModel.isBlockOfTypeOnPlane(forwardPosition, "lava", placementPlane)) {
       soundEffect = () => this.levelView.audioPlayer.play("fizz");
     }
-    const placedBlock = this.levelModel.placeBlockForward(blockType, placementPlane);
-    this.levelView.playPlaceBlockInFrontAnimation(this.levelModel.player.position, this.levelModel.player.facing, forwardPosition, placementPlane, placedBlock.blockType, () => {
+
+    this.levelView.playPlaceBlockInFrontAnimation(this.levelModel.player.position, this.levelModel.player.facing, forwardPosition, placementPlane, blockType, () => {
       this.levelModel.computeShadingPlane();
       this.levelModel.computeFowPlane();
       this.levelView.updateShadingPlane(this.levelModel.shadingPlane);
       this.levelView.updateFowPlane(this.levelModel.fowPlane);
       soundEffect();
+      this.levelModel.placeBlockForward(blockType, placementPlane);
```

Comment:
I'm seeing a small bug with `gameController.codeOrgAPI.placeInFront(null, 'grass', 'Player')`.  The newly-placed block doesn't have a shadow until another block is placed.

I think the fix is just to move this up above the call to `computeShadingPlane`.

Reply:
good catch! That's absolutely it. I didn't even notice the graphical discrepancy.